WinApi.Windows.Controls/Clipboard/Clipboard.cs

```diff
@@ -0,0 +1,143 @@
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private readonly Task _waitAction;
+        private readonly ClipboardFormat[] _clipboardFormats;
+        private DummyForm _wind;
+
+        public Clipboard()
+        {
+            _cancellationTokenSource = new CancellationTokenSource();
+            _waitAction = Task.Run(() => Bootstrap(), _cancellationTokenSource.Token);
+            _clipboardFormats = new[] { ClipboardFormat.CF_UNICODETEXT, ClipboardFormat.CF_TEXT, ClipboardFormat.CF_BITMAP};
+        }
+
+        private void Bootstrap()
+        {
+            using (_wind = Window.Create<DummyForm>("hidden"))
+            {
+                _wind.ClipboardUpdateCallback = async () => await Update();
+
+                if (User32Methods.AddClipboardFormatListener(_wind.Handle))
+                    new EventLoop().Run(_wind);
+                else
+                {
+                    throw new Exception("Can't subscribe to the clipboard");
+                }
+            }
+        }
+
+        private void ReleaseUnmanagedResources()
+        {
+            //TODO: implement cancelation token
+            _cancellationTokenSource.Cancel();
+            if (!_wind.IsDisposed)
+            {
+                _wind.Dispose();
+            }
+        }
+
+        public void Dispose()
+        {
+            ReleaseUnmanagedResources();
+            GC.SuppressFinalize(this);
+        }
+
+        ~Clipboard()
+        {
+            ReleaseUnmanagedResources();
+        }
+
+        public IClipboardData GetData(ClipboardFormat format)
+        {
+            switch (format)
+            {
+                // https://msdn.microsoft.com/en-us/library/windows/desktop/dd317756(v=vs.85).aspx
+                // ANSI code pages can be different on different computers, or can be changed for a single computer, leading to data corruption.
+                // For the most consistent results, applications should use Unicode, such as UTF-8 or UTF-16, instead of a specific code page.
+                case ClipboardFormat.CF_TEXT:
+                    return new TextData
+                    {
+                        Data = Encoding.Unicode.GetBytes(User32Helpers.GetTextFromClipboard())
+                    };
+                case ClipboardFormat.CF_UNICODETEXT:
+                    return new TextData
+                    {
+                        Data = Encoding.Unicode.GetBytes(User32Helpers.GetUnicodeTextFromClipboard())
+                    };
+                case ClipboardFormat.CF_BITMAP:
+                    byte[] data = null;
+                    try
+                    {
+                        if (User32Methods.OpenClipboard(new IntPtr()))
+                        {
+                            var handle = User32Methods.GetClipboardData((uint) ClipboardFormat.CF_BITMAP);
+                            data = GdiplusHelpers.CreateBitmap(handle) ?? throw new Exception("Can't read the bitmap");
```

Comment:
probably should throw bounded to context exceptions, like `ClipboardReadException`?